specfile/prep.py

```diff
@@ -112,19 +112,26 @@ class PatchMacro(PrepMacro):
     @property
     def number(self) -> int:
         """Number of the %patch macro."""
-        if self.options.P is not None:
-            return int(self.options.P)
         tokens = re.split(r"(\d+)", self.name, maxsplit=1)
         if len(tokens) > 1:
             return int(tokens[1])
```

Comment:
Can the `-P` option be in other than the first place? Cause I have a feeling that for this:
```
%patch -p2 -P28 -b .patch28
```

It would yield the `2` instead of the `28` :eyes:

Reply:
It can, but I'm confused - these two lines are for when the number is part of macro name, e.g. `%patch4`.
It is checked first because that's the priority of expansion: number in the name (with RPM < 4.20), number as the `-P` option and finally number as a positional argument.

Reply:
> It would yield the `2` instead of the `28` 👀

It wouldn't, `-p` is not `-P` and order doesn't matter.

Reply:
Oh, that's on the `.name`, not the whole line… got it